Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx

```diff
@@ -32,6 +33,25 @@ const SubmissionGridRow = ({ submission }: ISubmissionGridRowProps) => {
         isOfficial,
     } = submission;
 
+    const { state: loggedInUser } = useAuth();
+
+    const renderDetailsBtn = useCallback(
+        () => {
+            const { user: { username: loggedInUsername, permissions: { canAccessAdministration } } } = loggedInUser;
+
+            if (username === loggedInUsername || canAccessAdministration) {
```

Comment:
Can't this be moved as a helper method in auth and used here?

Also this should be moved on page level so we can redirect to the login page from there.

Reply:
I am not sure I get it. We are not checking if the user is logged in, but if the logged in user is the one that owns the submission or if the logged in user has administrative rights. In both those cases the details button is rendered.